### PR TITLE
Do Not Set `Name` and `Namespace` Tags on EB Environment

### PR DESCRIPTION
**Pull request by tchernomax**
## what

* Do not set `Name` and `Namespace` on Elastic Beanstalk Environment

## why
* When we try to set the tags "Name" and "Namespace" the second deployement fails (the first one is ok).

## example

```
* module.front-elastic-beanstalk-environment.aws_elastic_beanstalk_environment.default: 1 error(s) occurred:

* aws_elastic_beanstalk_environment.default: Error waiting for Elastic Beanstalk Environment (e-pkzfqk3igp) to become ready: 2 error(s) occurred:

* 2018-06-27 16:54:27.56 +0000 UTC (e-pkzfqk3igp) : Service:AmazonCloudFormation, Message:No updates are to be performed.
* 2018-06-27 16:54:28.178 +0000 UTC (e-pkzfqk3igp) : Environment tag update failed.
```

## references
- <https://github.com/terraform-providers/terraform-provider-aws/issues/3963>

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -341,7 +341,7 @@ resource "aws_elastic_beanstalk_environment" "default" {
 
   version_label = "${var.version_label}"
 
-  tags = "${module.label.tags}"
+  tags = "${merge(map("Stage", "${module.label.stage}"), var.tags)}"
```
Comment (osterman) [inline main.tf]: Hrmm... can you elaborate on how/why this appears to fix it?

**Comment by guikcd** [comment]
Beware that awsbeanstalk return an error when you try to submit same tags as currently set.

File: main.tf (hunk 1 above)
Comment (tchernomax) [reply main.tf]: @osterman On AWS console the tag "Name" is grayed, we can't change it.
I don't know why "Namespace" raise an error… also setting it by hand is ok.

module.label.tags = https://github.com/cloudposse/terraform-null-label/blob/master/main.tf#L9  
So module.label.tags = Name + Namespace + Stage + var.tags  
`${merge(map("Stage", "${module.label.stage}"), var.tags)}` is exactly the same without Name & Namespace.

**Comment by jaypea** [comment]
this is the relevat bug report at terraform: https://github.com/terraform-providers/terraform-provider-aws/issues/3963

**Comment by jaypea** [comment]
I've used this branch in my environment like this:
```
module "elastic-beanstalk-environment" {
  source  = "git::https://github.com/tchernomax/terraform-aws-elastic-beanstalk-environment.git?ref=tags"
```
and can confirm that it fixes the issue.

File: main.tf (hunk 1 above)
Comment (osterman) [reply main.tf]: Ok, I understand now. Thanks! We'll merge this today.

**Comment by osterman** [comment]
@jaypea thanks!

**Comment by jaypea** [comment]
@osterman i've tried to use `lifecycle { ignore_changes = ["tags"]}` and it works as well.
its the better solution in my oppion. its more robust to other changes in the tags list.

**Comment by osterman** [comment]
@jaypea if you have time to open a PR for that, I will approve. @tchernomax seems to be away.

**Comment by jaypea** [comment]
@osterman done, see #40

**Pull request closed without merging**